diff --git a/configure.ac b/configure.ac
index 562d621..f62703a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -22,7 +22,7 @@ AC_ARG_ENABLE([hardening],
   AS_HELP_STRING([--disable-hardening], [Disable hardening]))
 
 AS_IF([test "x$enable_hardening" != "xno"],
-      [EXTRA_CFLAGS="-Wall -Werror -D_FORTIFY_SOURCE=2"],
+      [EXTRA_CFLAGS="-Wall -Werror -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2"],
       [EXTRA_CFLAGS="-Wall -Werror"])
 
 AC_SUBST(EXTRA_CFLAGS)
