diff -Naur a/api/opencl/amdocl/glibc_functions.cpp b/api/opencl/amdocl/glibc_functions.cpp
--- a/api/opencl/amdocl/glibc_functions.cpp	2019-07-09 02:19:44.000000000 +0000
+++ b/api/opencl/amdocl/glibc_functions.cpp	2019-08-26 23:06:46.597420390 +0000
@@ -7,7 +7,6 @@
 #endif // __cplusplus
 
 #if defined(_LP64)
-asm (".symver memcpy, memcpy@GLIBC_2.2.5");
 void *__wrap_memcpy(void *dest, const void *src, size_t n)
 {
     return memcpy(dest, src, n);
