diff -Naur a/compiler/lib/loaders/elf/utils/libelf/_libelf.h b/compiler/lib/loaders/elf/utils/libelf/_libelf.h
--- a/compiler/lib/loaders/elf/utils/libelf/_libelf.h	2019-07-28 07:49:06.930140842 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/_libelf.h	2019-07-28 08:20:08.533501727 +0000
@@ -179,7 +179,10 @@
  * Function Prototypes.
  */
 
-__BEGIN_DECLS
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 Elf_Data *_libelf_allocate_data(Elf_Scn *_s);
 Elf	*_libelf_allocate_elf(Elf_Mem *mem);
 Elf_Scn	*_libelf_allocate_scn(Elf *_e, size_t _ndx);
@@ -213,6 +216,9 @@
 Elf_Data *_libelf_xlate(Elf_Data *_d, const Elf_Data *_s,
     unsigned int _encoding, int _elfclass, int _direction);
 int	_libelf_xlate_shtype(uint32_t _sht);
-__END_DECLS
+
+#ifdef __cplusplus
+}
+#endif
 
 #endif	/* __LIBELF_H_ */
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/elf_begin.c b/compiler/lib/loaders/elf/utils/libelf/elf_begin.c
--- a/compiler/lib/loaders/elf/utils/libelf/elf_begin.c	2019-07-28 07:49:06.930140842 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/elf_begin.c	2019-07-28 08:20:08.536835060 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <sys/types.h>
 #if !defined(WIN32)
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/elf_end.c b/compiler/lib/loaders/elf/utils/libelf/elf_end.c
--- a/compiler/lib/loaders/elf/utils/libelf/elf_end.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/elf_end.c	2019-07-28 08:20:08.536835060 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #if !defined(WIN32)
 #include <sys/mman.h>
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/elf_errmsg.c b/compiler/lib/loaders/elf/utils/libelf/elf_errmsg.c
--- a/compiler/lib/loaders/elf/utils/libelf/elf_errmsg.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/elf_errmsg.c	2019-07-28 08:20:08.536835060 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <libelf.h>
 #include <stdio.h>
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/elf_errno.c b/compiler/lib/loaders/elf/utils/libelf/elf_errno.c
--- a/compiler/lib/loaders/elf/utils/libelf/elf_errno.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/elf_errno.c	2019-07-28 08:20:08.536835060 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <libelf.h>
 
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/elf_fill.c b/compiler/lib/loaders/elf/utils/libelf/elf_fill.c
--- a/compiler/lib/loaders/elf/utils/libelf/elf_fill.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/elf_fill.c	2019-07-28 08:20:08.536835060 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <libelf.h>
 
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/elf_flag.c b/compiler/lib/loaders/elf/utils/libelf/elf_flag.c
--- a/compiler/lib/loaders/elf/utils/libelf/elf_flag.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/elf_flag.c	2019-07-28 08:20:08.536835060 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <libelf.h>
 
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/elf_getarhdr.c b/compiler/lib/loaders/elf/utils/libelf/elf_getarhdr.c
--- a/compiler/lib/loaders/elf/utils/libelf/elf_getarhdr.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/elf_getarhdr.c	2019-07-28 08:20:08.536835060 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <libelf.h>
 
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/elf_getarsym.c b/compiler/lib/loaders/elf/utils/libelf/elf_getarsym.c
--- a/compiler/lib/loaders/elf/utils/libelf/elf_getarsym.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/elf_getarsym.c	2019-07-28 08:20:08.536835060 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <libelf.h>
 
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/elf_getbase.c b/compiler/lib/loaders/elf/utils/libelf/elf_getbase.c
--- a/compiler/lib/loaders/elf/utils/libelf/elf_getbase.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/elf_getbase.c	2019-07-28 08:20:08.536835060 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <libelf.h>
 
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/elf_getident.c b/compiler/lib/loaders/elf/utils/libelf/elf_getident.c
--- a/compiler/lib/loaders/elf/utils/libelf/elf_getident.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/elf_getident.c	2019-07-28 08:20:08.536835060 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <ar.h>
 #include <assert.h>
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/elf_hash.c b/compiler/lib/loaders/elf/utils/libelf/elf_hash.c
--- a/compiler/lib/loaders/elf/utils/libelf/elf_hash.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/elf_hash.c	2019-07-28 08:20:08.536835060 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <libelf.h>
 
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/elf_kind.c b/compiler/lib/loaders/elf/utils/libelf/elf_kind.c
--- a/compiler/lib/loaders/elf/utils/libelf/elf_kind.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/elf_kind.c	2019-07-28 08:20:08.536835060 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <libelf.h>
 
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/elf_memory.c b/compiler/lib/loaders/elf/utils/libelf/elf_memory.c
--- a/compiler/lib/loaders/elf/utils/libelf/elf_memory.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/elf_memory.c	2019-07-28 08:20:08.536835060 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <ar.h>
 #include <libelf.h>
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/elf_next.c b/compiler/lib/loaders/elf/utils/libelf/elf_next.c
--- a/compiler/lib/loaders/elf/utils/libelf/elf_next.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/elf_next.c	2019-07-28 08:20:08.536835060 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <ar.h>
 #include <assert.h>
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/elf_phnum.c b/compiler/lib/loaders/elf/utils/libelf/elf_phnum.c
--- a/compiler/lib/loaders/elf/utils/libelf/elf_phnum.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/elf_phnum.c	2019-07-28 08:20:08.540168394 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <ar.h>
 #include <libelf.h>
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/elf_rand.c b/compiler/lib/loaders/elf/utils/libelf/elf_rand.c
--- a/compiler/lib/loaders/elf/utils/libelf/elf_rand.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/elf_rand.c	2019-07-28 08:20:08.540168394 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <ar.h>
 #include <libelf.h>
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/elf_rawfile.c b/compiler/lib/loaders/elf/utils/libelf/elf_rawfile.c
--- a/compiler/lib/loaders/elf/utils/libelf/elf_rawfile.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/elf_rawfile.c	2019-07-28 08:20:08.540168394 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <libelf.h>
 
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/elf_scn.c b/compiler/lib/loaders/elf/utils/libelf/elf_scn.c
--- a/compiler/lib/loaders/elf/utils/libelf/elf_scn.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/elf_scn.c	2019-07-28 08:20:08.540168394 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 #include <sys/queue.h>
 
 #include <assert.h>
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/elf_shnum.c b/compiler/lib/loaders/elf/utils/libelf/elf_shnum.c
--- a/compiler/lib/loaders/elf/utils/libelf/elf_shnum.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/elf_shnum.c	2019-07-28 08:20:08.540168394 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <ar.h>
 #include <libelf.h>
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/elf_shstrndx.c b/compiler/lib/loaders/elf/utils/libelf/elf_shstrndx.c
--- a/compiler/lib/loaders/elf/utils/libelf/elf_shstrndx.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/elf_shstrndx.c	2019-07-28 08:20:08.540168394 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <ar.h>
 #include <libelf.h>
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/elf_version.c b/compiler/lib/loaders/elf/utils/libelf/elf_version.c
--- a/compiler/lib/loaders/elf/utils/libelf/elf_version.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/elf_version.c	2019-07-28 08:20:08.540168394 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <libelf.h>
 
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/gelf.h b/compiler/lib/loaders/elf/utils/libelf/gelf.h
--- a/compiler/lib/loaders/elf/utils/libelf/gelf.h	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/gelf.h	2019-07-28 08:20:08.540168394 +0000
@@ -29,7 +29,6 @@
 #ifndef	_GELF_H_
 #define	_GELF_H_
 
-#include <sys/cdefs.h>
 
 #include <libelf.h>
 
@@ -72,7 +71,10 @@
 #define	GELF_ST_TYPE			ELF64_ST_TYPE
 #define	GELF_ST_VISIBILITY		ELF64_ST_VISIBILITY
 
-__BEGIN_DECLS
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 long		gelf_checksum(Elf *_elf);
 size_t		gelf_fsize(Elf *_elf, Elf_Type _type, size_t _count,
 			unsigned int _version);
@@ -109,6 +111,9 @@
 int		gelf_update_move(Elf_Data *_dst, int _index, GElf_Move *_src);
 int		gelf_update_syminfo(Elf_Data *_dst, int _index, GElf_Syminfo *_src);
 #endif
-__END_DECLS
+
+#ifdef __cplusplus
+}
+#endif
 
 #endif	/* _GELF_H_ */
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/gelf_cap.c b/compiler/lib/loaders/elf/utils/libelf/gelf_cap.c
--- a/compiler/lib/loaders/elf/utils/libelf/gelf_cap.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/gelf_cap.c	2019-07-28 08:20:08.540168394 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <assert.h>
 #include <gelf.h>
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/gelf_checksum.c b/compiler/lib/loaders/elf/utils/libelf/gelf_checksum.c
--- a/compiler/lib/loaders/elf/utils/libelf/gelf_checksum.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/gelf_checksum.c	2019-07-28 08:20:08.540168394 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <gelf.h>
 #include <libelf.h>
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/gelf_dyn.c b/compiler/lib/loaders/elf/utils/libelf/gelf_dyn.c
--- a/compiler/lib/loaders/elf/utils/libelf/gelf_dyn.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/gelf_dyn.c	2019-07-28 08:20:08.540168394 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <assert.h>
 #include <gelf.h>
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/gelf_ehdr.c b/compiler/lib/loaders/elf/utils/libelf/gelf_ehdr.c
--- a/compiler/lib/loaders/elf/utils/libelf/gelf_ehdr.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/gelf_ehdr.c	2019-07-28 08:20:08.540168394 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <assert.h>
 #include <gelf.h>
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/gelf_fsize.c b/compiler/lib/loaders/elf/utils/libelf/gelf_fsize.c
--- a/compiler/lib/loaders/elf/utils/libelf/gelf_fsize.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/gelf_fsize.c	2019-07-28 08:20:08.540168394 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <gelf.h>
 #include <libelf.h>
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/gelf_getclass.c b/compiler/lib/loaders/elf/utils/libelf/gelf_getclass.c
--- a/compiler/lib/loaders/elf/utils/libelf/gelf_getclass.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/gelf_getclass.c	2019-07-28 08:20:08.540168394 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <gelf.h>
 
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/gelf_move.c b/compiler/lib/loaders/elf/utils/libelf/gelf_move.c
--- a/compiler/lib/loaders/elf/utils/libelf/gelf_move.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/gelf_move.c	2019-07-28 08:20:08.543501727 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <assert.h>
 #include <gelf.h>
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/gelf_phdr.c b/compiler/lib/loaders/elf/utils/libelf/gelf_phdr.c
--- a/compiler/lib/loaders/elf/utils/libelf/gelf_phdr.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/gelf_phdr.c	2019-07-28 08:20:08.543501727 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <gelf.h>
 #include <libelf.h>
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/gelf_rel.c b/compiler/lib/loaders/elf/utils/libelf/gelf_rel.c
--- a/compiler/lib/loaders/elf/utils/libelf/gelf_rel.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/gelf_rel.c	2019-07-28 08:20:08.543501727 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <assert.h>
 #include <gelf.h>
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/gelf_rela.c b/compiler/lib/loaders/elf/utils/libelf/gelf_rela.c
--- a/compiler/lib/loaders/elf/utils/libelf/gelf_rela.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/gelf_rela.c	2019-07-28 08:20:08.543501727 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <assert.h>
 #include <gelf.h>
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/gelf_shdr.c b/compiler/lib/loaders/elf/utils/libelf/gelf_shdr.c
--- a/compiler/lib/loaders/elf/utils/libelf/gelf_shdr.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/gelf_shdr.c	2019-07-28 08:20:08.543501727 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <assert.h>
 #include <gelf.h>
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/gelf_sym.c b/compiler/lib/loaders/elf/utils/libelf/gelf_sym.c
--- a/compiler/lib/loaders/elf/utils/libelf/gelf_sym.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/gelf_sym.c	2019-07-28 08:20:08.543501727 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <assert.h>
 #include <gelf.h>
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/gelf_syminfo.c b/compiler/lib/loaders/elf/utils/libelf/gelf_syminfo.c
--- a/compiler/lib/loaders/elf/utils/libelf/gelf_syminfo.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/gelf_syminfo.c	2019-07-28 08:20:08.543501727 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <assert.h>
 #include <gelf.h>
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/gelf_symshndx.c b/compiler/lib/loaders/elf/utils/libelf/gelf_symshndx.c
--- a/compiler/lib/loaders/elf/utils/libelf/gelf_symshndx.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/gelf_symshndx.c	2019-07-28 08:20:08.543501727 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <assert.h>
 #include <gelf.h>
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/gelf_xlate.c b/compiler/lib/loaders/elf/utils/libelf/gelf_xlate.c
--- a/compiler/lib/loaders/elf/utils/libelf/gelf_xlate.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/gelf_xlate.c	2019-07-28 08:20:08.543501727 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <gelf.h>
 #include <libelf.h>
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/libelf.h b/compiler/lib/loaders/elf/utils/libelf/libelf.h
--- a/compiler/lib/loaders/elf/utils/libelf/libelf.h	2019-07-28 07:56:13.316813863 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/libelf.h	2019-07-28 08:20:08.543501727 +0000
@@ -187,7 +187,10 @@
 #define	ELF_F_ARCHIVE	   0x100U /* archive creation */
 #define	ELF_F_ARCHIVE_SYSV 0x200U /* SYSV style archive */
 
-__BEGIN_DECLS
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 Elf		*elf_begin(int _fd, Elf_Cmd _cmd, Elf *_elf, Elf_Mem *_mem);
 int		elf_cntl(Elf *_elf, Elf_Cmd _cmd);
 int		elf_end(Elf *_elf);
@@ -257,6 +260,8 @@
 			unsigned int _enc);
 Elf_Data	*elf64_xlatetom(Elf_Data *_dst, const Elf_Data *_src,
 			unsigned int _enc);
-__END_DECLS
+#ifdef __cplusplus
+}
+#endif
 
 #endif	/* _LIBELF_H_ */
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/libelf_align.c b/compiler/lib/loaders/elf/utils/libelf/libelf_align.c
--- a/compiler/lib/loaders/elf/utils/libelf/libelf_align.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/libelf_align.c	2019-07-28 08:20:08.543501727 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <sys/types.h>
 
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/libelf_allocate.c b/compiler/lib/loaders/elf/utils/libelf/libelf_allocate.c
--- a/compiler/lib/loaders/elf/utils/libelf/libelf_allocate.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/libelf_allocate.c	2019-07-28 08:20:08.543501727 +0000
@@ -28,7 +28,6 @@
  * Internal APIs
  */
 
-#include <sys/cdefs.h>
 
 #if !defined(WIN32)
 #include <sys/errno.h>
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/libelf_ar.c b/compiler/lib/loaders/elf/utils/libelf/libelf_ar.c
--- a/compiler/lib/loaders/elf/utils/libelf/libelf_ar.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/libelf_ar.c	2019-07-28 08:20:08.543501727 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <assert.h>
 #include <ctype.h>
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/libelf_ar_util.c b/compiler/lib/loaders/elf/utils/libelf/libelf_ar_util.c
--- a/compiler/lib/loaders/elf/utils/libelf/libelf_ar_util.c	2019-07-28 07:49:06.933474175 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/libelf_ar_util.c	2019-07-28 08:20:08.543501727 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <assert.h>
 #include <libelf.h>
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/libelf_checksum.c b/compiler/lib/loaders/elf/utils/libelf/libelf_checksum.c
--- a/compiler/lib/loaders/elf/utils/libelf/libelf_checksum.c	2019-07-28 07:49:06.936807509 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/libelf_checksum.c	2019-07-28 08:20:08.543501727 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <gelf.h>
 
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/libelf_convert.c b/compiler/lib/loaders/elf/utils/libelf/libelf_convert.c
--- a/compiler/lib/loaders/elf/utils/libelf/libelf_convert.c	2019-07-28 07:49:06.936807509 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/libelf_convert.c	2019-07-28 08:20:08.546835061 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <assert.h>
 #include <libelf.h>
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/libelf_data.c b/compiler/lib/loaders/elf/utils/libelf/libelf_data.c
--- a/compiler/lib/loaders/elf/utils/libelf/libelf_data.c	2019-07-28 07:49:06.936807509 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/libelf_data.c	2019-07-28 08:20:08.546835061 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <libelf.h>
 
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/libelf_ehdr.c b/compiler/lib/loaders/elf/utils/libelf/libelf_ehdr.c
--- a/compiler/lib/loaders/elf/utils/libelf/libelf_ehdr.c	2019-07-28 07:49:06.936807509 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/libelf_ehdr.c	2019-07-28 08:20:08.546835061 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <assert.h>
 #include <gelf.h>
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/libelf_extended.c b/compiler/lib/loaders/elf/utils/libelf/libelf_extended.c
--- a/compiler/lib/loaders/elf/utils/libelf/libelf_extended.c	2019-07-28 07:49:06.936807509 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/libelf_extended.c	2019-07-28 08:20:08.546835061 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <assert.h>
 #include <libelf.h>
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/libelf_msize.c b/compiler/lib/loaders/elf/utils/libelf/libelf_msize.c
--- a/compiler/lib/loaders/elf/utils/libelf/libelf_msize.c	2019-07-28 07:49:06.936807509 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/libelf_msize.c	2019-07-28 08:20:08.546835061 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <assert.h>
 #include <libelf.h>
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/libelf_phdr.c b/compiler/lib/loaders/elf/utils/libelf/libelf_phdr.c
--- a/compiler/lib/loaders/elf/utils/libelf/libelf_phdr.c	2019-07-28 07:49:06.936807509 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/libelf_phdr.c	2019-07-28 08:20:08.546835061 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <assert.h>
 #include <gelf.h>
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/libelf_shdr.c b/compiler/lib/loaders/elf/utils/libelf/libelf_shdr.c
--- a/compiler/lib/loaders/elf/utils/libelf/libelf_shdr.c	2019-07-28 07:49:06.936807509 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/libelf_shdr.c	2019-07-28 08:20:08.546835061 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <gelf.h>
 #include <libelf.h>
diff -Naur a/compiler/lib/loaders/elf/utils/libelf/libelf_xlate.c b/compiler/lib/loaders/elf/utils/libelf/libelf_xlate.c
--- a/compiler/lib/loaders/elf/utils/libelf/libelf_xlate.c	2019-07-28 07:49:06.936807509 +0000
+++ b/compiler/lib/loaders/elf/utils/libelf/libelf_xlate.c	2019-07-28 08:20:08.546835061 +0000
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <assert.h>
 #include <libelf.h>
