diff -Naur a/src/libhsakmt.h b/src/libhsakmt.h
--- a/src/libhsakmt.h	2019-06-16 23:24:22.000000000 +0000
+++ b/src/libhsakmt.h	2019-07-28 07:03:17.256766568 +0000
@@ -49,6 +49,10 @@
 #define CHECK_KFD_OPEN() \
 	do { if (kfd_open_count == 0) return HSAKMT_STATUS_KERNEL_IO_CHANNEL_NOT_OPENED; } while (0)
 
+#ifdef PAGE_SIZE
+#undef PAGE_SIZE
+#endif
+
 extern int PAGE_SIZE;
 extern int PAGE_SHIFT;
 
