--- a/src/engine/backend/src/keycode.rs
+++ b/src/engine/backend/src/keycode.rs
@@ -229,7 +229,7 @@
             102 => Some(Self::Muhenkan),
             64 => Some(Self::AltL),
             108 => Some(Self::AltR),
-            122 | 130 => Some(Self::Hangul),
+            122 | 130 | 134 => Some(Self::Hangul),
             121 | 123 | 131 => Some(Self::HangulHanja),
 
             113 => Some(Self::Left),
